docs/examples/multi-agent-workflow.md

```diff
@@ -0,0 +1,184 @@
+                rules=[
+                    Rule(
+                        value="Lead Research Analyst",
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Objective",
+                rules=[
+                    Rule(
+                        value="Discover innovative advancements in artificial intelligence and data analytics",
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Background",
+                rules=[
+                    Rule(
+                        value="""You are part of a prominent technology research institute.
+                        Your speciality is spotting new trends.
+                        You excel at analyzing intricate data and delivering practical insights."""
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Desired Outcome",
+                rules=[
+                    Rule(
+                        value="Comprehensive analysis report in list format",
+                    )
+                ],
+            ),
+        ],
+    )
+
+    return researcher
+
+
+def build_writer(role: str, goal: str, backstory: str):
+    """Builds a Writer Structure.
+
+    Args:
+        role: The role of the writer.
+        goal: The goal of the writer.
+        backstory: The backstory of the writer.
+    """
+    writer = Agent(
+        id=role.lower().replace(" ", "_"),
+        event_listeners=[
+            EventListener(
+                event_types=[FinishStructureRunEvent],
+                driver=WebhookEventListenerDriver(
+                    webhook_url=os.environ["WEBHOOK_URL"],
+                ),
+            )
+        ],
+        rulesets=[
+            Ruleset(
+                name="Position",
+                rules=[
+                    Rule(
+                        value=role,
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Objective",
+                rules=[
+                    Rule(
+                        value=goal,
+                    )
+                ],
+            ),
+            Ruleset(
+                name="Backstory",
+                rules=[Rule(value=backstory)],
+            ),
+            Ruleset(
+                name="Desired Outcome",
+                rules=[
+                    Rule(
+                        value="Full blog post of at least 4 paragraphs",
+                    )
+                ],
+            ),
+        ],
+    )
+
+    return writer
+
+
+if __name__ == "__main__":
+    # Build the team
+    team = Workflow()
+    research_task = team.add_task(
+        StructureRunTask(
+            """Conduct a comprehensive analysis of the latest advancements in AI in 2024.
+                Identify key trends, breakthrough technologies, and potential industry impacts.""",
+            id="research",
+            target_structure=build_researcher(),
```

Comment:
Similar to `LocalStructureRunDriver`, `StructureRunTask` should accept a `structure_factory_fn` instead of an actual `Structure` instance to avoid the pitfalls of shared state between concurrent runs, right?

Reply:
This was actually a mistake in the examples, `target_structure` is no longer a field. The Tool takes a Driver just like the Task.